server/views/pages/createGoals/index.njk

```diff
@@ -123,28 +123,29 @@
                 errorMessage: errors | findError('targetCompletionDate')
               }) }}
 
-            <div class="govuk-form-group">
-              <!-- TODO: Populate the stepNumber object to relect the correct number on the UI -->
-              {% set stepTitleLabel = 'Step ' + form.stepNumber %}
-              <!-- TODO: Populate the stepTitle object to reflect the correct step title on the UI -->
-              {{ govukCharacterCount({
-                name: "stepTitle",
-                id: "stepTitle",
-                value: form.stepTitle,
-                maxlength: 512,
-                rows: 2,
-                label: {
-                  text: stepTitleLabel,
-                  classes: "govuk-label--s",
-                  isPageHeading: false
-                },
-                hint: {
-                  text: "Describe this step"
-                },
-                errorMessage: errors | findError('stepTitle')
-              }) }}
-
-            </div>
+            {% for step in form.steps %}
+              <div class="govuk-form-group">
+                <!-- TODO: Populate the stepNumber object to relect the correct number on the UI -->
+                {% set stepTitleLabel = 'Step ' + form.stepNumber %}
+                <!-- TODO: Populate the stepTitle object to reflect the correct step title on the UI -->
+                {{ govukCharacterCount({
+                  name: "stepTitle",
+                  id: "stepTitle",
+                  value: form.goals[0].steps[0].title,
+                  maxlength: 512,
+                  rows: 2,
+                  label: {
+                    text: stepTitleLabel,
+                    classes: "govuk-label--s",
+                    isPageHeading: false
+                  },
+                  hint: {
+                    text: "Describe this step"
+                  },
+                  errorMessage: errors | findError('stepTitle')
```

Comment:
`findError` will need to reference the `id` of _this_ step title field, so will want to be something like 
```
errorMessage: errors | findError('goals[0].step[' + loop.index0 + '].title')
```